base/habana-python-3.10/Dockerfile

```diff
@@ -0,0 +1,80 @@
+FROM vault.habana.ai/gaudi-docker/1.15.1/rhel9.2/habanalabs/pytorch-installer-2.2.0:1.15.1-15 as base
+
+LABEL name="odh-notebook-habana-jupyter-1.15.1-ubi9-python-3.10" \
+    summary="Jupyter HabanaAI 1.15.1 notebook image for ODH notebooks" \
+    description="Jupyter HabanaAI 1.15.1 notebook image with base Python 3.10 builder image based on ubi9 for ODH notebooks" \
+    io.k8s.display-name="Jupyter HabanaAI 1.15.1 notebook image for ODH notebooks" \
+    io.k8s.description="Jupyter HabanaAI 1.15.1 notebook image with base Python 3.10 builder image based on ubi9 for ODH notebooks" \
+    authoritative-source-url="https://github.com/opendatahub-io/notebooks" \
+    io.openshift.build.commit.ref="main" \
+    io.openshift.build.source-location="https://github.com/opendatahub-io/notebooks/tree/main/habana/1.15.1/ubi9-python-3.10" \
+    io.openshift.build.image="quay.io/opendatahub/workbench-images:habana-jupyter-1.15.1-ubi9-python-3.10"
+
+COPY --from=registry.access.redhat.com/ubi9/python-39:latest /usr/bin/fix-permissions /usr/bin
+COPY --from=registry.access.redhat.com/ubi9/python-39:latest /usr/bin/rpm-file-permissions /usr/bin
+
+ENV PATH=/opt/app-root/src/bin:/opt/app-root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
+    APP_ROOT="/opt/app-root"
+
+ENV HOME="${APP_ROOT}/src"
+
+WORKDIR ${HOME}
+
+RUN mkdir -p ${APP_ROOT}/src && \
+    rpm-file-permissions && \
+    useradd -u 1001 -r -g 0 -d ${HOME} -c "Default Application User" default && \
+    chown -R 1001:0 ${APP_ROOT}
+
+FROM base as notebook-base
+
+ENV PYTHON_VERSION=3.10 \
+    PATH=$HOME/.local/bin/:$PATH \
+    PYTHONUNBUFFERED=1 \
+    PYTHONIOENCODING=UTF-8 \
+    LC_ALL=en_US.UTF-8 \
+    LANG=en_US.UTF-8 \
+    PIP_NO_CACHE_DIR=off
+
+RUN python3.10 -m venv --system-site-packages ${APP_ROOT} && \
+    chown -R 1001:0 ${APP_ROOT} && \
+    fix-permissions ${APP_ROOT} -P && \
+    rpm-file-permissions && \
+    echo "unset BASH_ENV PROMPT_COMMAND ENV" >> ${APP_ROOT}/bin/activate
+
+ENV BASH_ENV="${APP_ROOT}/bin/activate" \
+    ENV="${APP_ROOT}/bin/activate" \
+    PROMPT_COMMAND=". ${APP_ROOT}/bin/activate"
+
+# Squash a warning from the Habana runtime
+ENV TSAN_OPTIONS='ignore_noninstrumented_modules=1'
```

Comment:
are we really compiling something with thread sanitizer enabled? this gives you 3 to 10x more memory consumption and about the same amount of performance degradation; tsan is intended for debug builds only